src/ol/source/GeoTIFF.js

```diff
@@ -382,7 +384,7 @@ class GeoTIFFSource extends DataTile {
     super({
       state: 'loading',
       tileGrid: null,
-      projection: null,
+      projection: options.projection || null,
```

Comment:
I've extracted the changes in this file to #15036 so that we could also release this PR separately, if required.